src/note_position.rs

```diff
@@ -0,0 +1,92 @@
+use std::fmt;
+use std::iter::Peekable;
+
+/// An iterator adaptor that wraps each element in an [`Position`](../enum.Position.html).
+///
+/// Iterator element type is `Position<I::Item>`.
+/// This iterator is *fused*.
```

Comment:
A fused iterator would pass the test called `exact_position` in this diff on top of your PR (as it was), and it doesn't. Peekable calls `.next()` on the underlying iterator, even if it has seen `None` already.

```rust
diff --git a/src/note_position.rs b/src/note_position.rs
index 58f8dc8..8bac7b1 100644
--- a/src/note_position.rs
+++ b/src/note_position.rs
@@ -89,4 +89,8 @@ impl<I: Iterator> Iterator for NotePosition<I> {
             (_, None) => None,
         }
     }
+
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        self.peekable.size_hint()
+    }
 }
diff --git a/tests/quick.rs b/tests/quick.rs
index a0176c8..9aa8681 100644
--- a/tests/quick.rs
+++ b/tests/quick.rs
@@ -346,6 +346,12 @@ quickcheck! {
     fn exact_interleave(a: Iter<i16>, b: Iter<i16>) -> bool {
         exact_size_for_this(a.interleave(b))
     }
+    fn exact_position(a: Iter<i16>) -> bool {
+        exact_size_for_this(a.note_position())
+    }
+    fn exact_position_2(a: Vec<i16>) -> bool {
+        exact_size_for_this(a.iter().note_position())
+    }
     fn size_interleave_shortest(a: Iter<i16>, b: Iter<i16>) -> bool {
         correct_size_hint(a.interleave_shortest(b))
     }
```

(`Iter` is a non-fused iterator that the quickcheck test suite uses).